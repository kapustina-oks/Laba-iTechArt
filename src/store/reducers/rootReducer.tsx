import { combineReducers } from "redux";
import authReducer from "./authReducer";
import adminReducer from "./adminReducer";
import cartReducer from "./cartReducer";

const rootReducer = combineReducers({
  auth: authReducer,
  admin: adminReducer,
  cart: cartReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
