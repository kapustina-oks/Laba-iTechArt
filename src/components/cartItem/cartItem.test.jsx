import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import CartItem from "./cartItem";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "../../store/reducers/rootReducer";

import dataGames from "../../mock/dataBase";

describe("CardCategory", () => {
  const store = configureStore({ reducer: rootReducer });

  it("testing of cart item", () => {
    render(<Provider store={store}><CartItem game={dataGames[0]}/></Provider>);
  });

  it("renders contain some text", () => {
    render(<Provider store={store}><CartItem game={dataGames[0]} /></Provider>);
    expect(screen.getByText("Game of Thrones")).toBeInTheDocument();
  });

  it("CartItem snapshot", () => {
    const card = render(<Provider store={store}><CartItem game={dataGames[0]} /></Provider>);
    expect(document.body).toMatchSnapshot();
  });
});
