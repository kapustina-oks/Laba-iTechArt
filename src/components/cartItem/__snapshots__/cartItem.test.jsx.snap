// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCategory CartItem snapshotg 1`] = `
<body>
  <div>
    <tr>
      <td
        class="cart-title"
      >
        Game of Thrones
      </td>
      <td>
        <select
          class="cart-select"
          name="select"
        >
          <option
            value="value3"
          >
            playstation
          </option>
          <option
            value="value3"
          >
            pc
          </option>
          <option
            value="value3"
          >
            xbox
          </option>
        </select>
      </td>
      <td
        class="cart-title"
      >
        12/28/2021
      </td>
      <td
        class="cart-title"
      >
        <input
          class="cart-input"
          id="qty"
          min="1"
          name="qty"
          type="number"
          value=""
        />
      </td>
      <td
        class="cart-title"
      >
        NaN
        $
      </td>
      <td>
        <button>
          <i
            class="fas fa-trash-alt cart-icon"
          />
        </button>
      </td>
    </tr>
  </div>
</body>
`;
